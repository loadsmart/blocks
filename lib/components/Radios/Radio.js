"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_native_1 = require("react-native");
var res_1 = require("../../res");
var Radio = /** @class */ (function (_super) {
    __extends(Radio, _super);
    function Radio() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Radio.prototype.render = function () {
        var isSelected = this.props.selected === true;
        var color = isSelected ? res_1.Colors.AlgaeGreen : res_1.Colors.Tuna;
        return (react_1.default.createElement(react_native_1.View, { style: [styles.outerCircle, this.props.style, { borderColor: color }] }, isSelected && react_1.default.createElement(react_native_1.View, { style: [styles.innerCircle, { backgroundColor: color }] })));
    };
    return Radio;
}(react_1.PureComponent));
exports.default = Radio;
var styles = react_native_1.StyleSheet.create({
    outerCircle: {
        width: 20,
        height: 20,
        borderWidth: 2,
        borderRadius: 10,
        alignItems: 'center',
        justifyContent: 'center',
    },
    innerCircle: {
        width: 12,
        height: 12,
        borderRadius: 6,
    },
});
