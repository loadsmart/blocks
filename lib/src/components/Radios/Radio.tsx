import React, { PureComponent } from 'react'
import { StyleSheet, View, ViewProps } from 'react-native'
import { Colors } from '../../res'

interface Props extends ViewProps {
  selected?: boolean
}

export default class Radio extends PureComponent<Props> {
  public render() {
    const isSelected = this.props.selected === true
    const color = isSelected ? Colors.AlgaeGreen : Colors.Tuna
    return (
      <View style={[styles.outerCircle, { borderColor: color }]}>
        {isSelected && <View style={[styles.innerCircle, { backgroundColor: color }]} />}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  outerCircle: {
    width: 20,
    height: 20,
    borderWidth: 2,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  innerCircle: {
    width: 12,
    height: 12,
    borderRadius: 6,
  },
})
