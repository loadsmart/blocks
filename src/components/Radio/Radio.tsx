import React, { PureComponent } from 'react'
import { View } from 'react-native'

export class Radio extends PureComponent {
  render() {
    return <View />
  }
}
