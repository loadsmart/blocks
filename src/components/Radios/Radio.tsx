import React, { PureComponent } from 'react'
import { StyleSheet, View, ViewProps } from 'react-native'
import { Colors } from '../../res'
import { ThemeContext } from '../Contexts'

interface Props extends ViewProps {
  selected?: boolean
}

export default class Radio extends PureComponent<Props> {
  public render() {
    const { selected, testID } = this.props
    return (
      <ThemeContext.Consumer>
        {theme => {
          const isSelected = selected === true
          const color = isSelected ? theme.primaryColor : Colors.Tuna
          return (
            <View
              style={[styles.outerCircle, this.props.style, { borderColor: color }]}
              {...{ testID }}
            >
              {isSelected && <View style={[styles.innerCircle, { backgroundColor: color }]} />}
            </View>
          )
        }}
      </ThemeContext.Consumer>
    )
  }
}

const styles = StyleSheet.create({
  outerCircle: {
    width: 20,
    height: 20,
    borderWidth: 2,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  innerCircle: {
    width: 12,
    height: 12,
    borderRadius: 6,
  },
})
