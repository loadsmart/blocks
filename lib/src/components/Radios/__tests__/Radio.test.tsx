import React from 'react'
import renderer from 'react-test-renderer'
import Radio from '../Radio'

describe('Radio', () => {
  it('matches snapshot when selected', () => {
    const testRenderer = renderer.create(<Radio selected={true} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot when not selected', () => {
    const testRenderer = renderer.create(<Radio selected={false} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
