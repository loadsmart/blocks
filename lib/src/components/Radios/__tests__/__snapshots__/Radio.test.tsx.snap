// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio matches snapshot when not selected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "borderColor": "#484a4c",
      },
    ]
  }
/>
`;

exports[`Radio matches snapshot when selected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 10,
        "borderWidth": 2,
        "height": 20,
        "justifyContent": "center",
        "width": 20,
      },
      Object {
        "borderColor": "#14d64d",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 6,
          "height": 12,
          "width": 12,
        },
        Object {
          "backgroundColor": "#14d64d",
        },
      ]
    }
  />
</View>
`;
